src/Smalot/PdfParser/RawData/RawDataParser.php

```diff
@@ -762,17 +755,18 @@ protected function getRawObject(string $pdfData, int $offset = 0): array
                     // start stream object
                     $objtype = 'stream';
                     $offset += 6;
-                    if (1 == preg_match('/^([\r]?[\n])/isU', substr($pdfData, $offset), $matches)) {
+                    if (1 == preg_match('/^([\r]?[\n])/isU', substr($pdfData, $offset, 4), $matches)) {
```

Comment:
Can you give me a short hint, why you added `4` (= length of substring) here?

Reply:
I'm not strong with inner structure of php string-like objects, their byteness, whether they are null-terminated or not (looks like they are not).

Since this regexp can match up to 2 character, 4 is 2 char * up to 2 bytes/char.
Probably it should be 2 insead of 4.

Reply:
> I'm not strong with inner structure of php string-like objects, their byteness, whether they are null-terminated or not (looks like they are not).
Since this regexp can match up to 2 character, 4 is 2 char * up to 2 bytes/char.
Probably it should be 2 insead of 4.

I am not sure if I can follow your argument. The additional `4` is part of this pull request, why was it added? I don't want to split hairs here, just to make sure there are no drive-by changes were not meant to be part of the PR in the end.

Reply:
Sorry, i was sure i have answered this question earlier:

Imagine this stream object appears in the beginning of 20M pdf. If so, this unlimited substr results in almost 20M RAM allocation.

And this regexp can match only 1 or 2 first characters of the string. 
That's why we can take no more than 4 bytes of $pdfData and match regexp against them.

Reply:
My bad, I overlooked it. Thanks for the clarification.